hadoop-ozone/integration-test/src/test/java/org/apache/hadoop/ozone/om/TestOmSnapshot.java

```diff
@@ -1786,6 +1787,46 @@ public void testDeleteSnapshotMissingMandatoryParams() throws Exception {
             () -> store.deleteSnapshot(nullstr, bucket, snap1));
   }
 
+  @Test
+  public void testSnapshotQuotaHandling() throws Exception {
+    String volume = "vol-" + counter.incrementAndGet();
+    String bucket = "buck-" + counter.incrementAndGet();
+    store.createVolume(volume);
+    OzoneVolume volume1 = store.getVolume(volume);
+    volume1.createBucket(bucket);
+    OzoneBucket bucket1 = volume1.getBucket(bucket);
+    bucket1.setQuota(OzoneQuota.parseQuota("102400000", "500"));
+    volume1.setQuota(OzoneQuota.parseQuota("204800000", "1000"));
+    // Create Key1 and take snapshot
+    String key1 = "key-1-";
+    createFileKeyWithPrefix(bucket1, key1);
+
+    long volNameQuotaBefore = volume1.getUsedNamespace();
+    long buckNameQuotaBefore = bucket1.getUsedNamespace();
+    long buckSpaceQuotaBefore = bucket1.getUsedBytes();
+
+    String snap1 = "snap" + counter.incrementAndGet();
+    createSnapshot(volume, bucket, snap1);
+
+    long volNameQuotaAfter = volume1.getUsedNamespace();
+    long buckNameQuotaAfter = bucket1.getUsedNamespace();
+    long buckSpaceQuotaAfter = bucket1.getUsedBytes();
+
+    assertEquals(volNameQuotaBefore, volNameQuotaAfter);
+    assertEquals(buckNameQuotaBefore, buckNameQuotaAfter);
+    assertEquals(buckSpaceQuotaBefore, buckSpaceQuotaAfter);
+
+    store.deleteSnapshot(volume, bucket, snap1);
+
+    long volNameQuotaFinal = volume1.getUsedNamespace();
+    long buckNameQuotaFinal = bucket1.getUsedNamespace();
+    long buckSpaceQuotaFinal = bucket1.getUsedBytes();
+
+    assertEquals(volNameQuotaBefore, volNameQuotaFinal);
+    assertEquals(buckNameQuotaBefore, buckNameQuotaFinal);
+    assertEquals(buckSpaceQuotaBefore, buckSpaceQuotaFinal);
```

Comment:
maybe add a case where key is deleted after taking a snapshot

Reply:
Added delete scenario